packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.derived.js

```diff
@@ -1,340 +1,392 @@
-import algoliaClient from 'algoliasearch/lite';
 import createInstantSearchManager from '../createInstantSearchManager';
 
-jest.useFakeTimers();
-
-jest.mock('algoliasearch-helper/src/algoliasearch.helper.js', () => {
-  const Helper = require.requireActual(
-    'algoliasearch-helper/src/algoliasearch.helper.js'
-  );
-  Helper.prototype._dispatchAlgoliaResponse = function(state) {
-    state.forEach(s => {
-      this.emit(
-        'result',
-        { query: s.state.query, page: s.state.page, index: s.state.index },
-        s
-      );
+const createSearchClient = () => ({
+  search(requests) {
+    return Promise.resolve({
+      results: requests.map(request => ({
+        index: request.indexName,
+        query: request.params.query,
+        page: request.params.page,
+        hits: [],
+      })),
     });
-  };
-  Helper.prototype.searchForFacetValues = () =>
-    Promise.resolve({ facetHits: 'results' });
-  return Helper;
+  },
+  searchForFacetValues() {
+    return Promise.resolve([
+      {
+        facetHits: [],
+      },
+    ]);
+  },
 });
 
-const client = algoliaClient('latency', '249078a3d4337a8231f1665ec5a44966');
-client.search = jest.fn(() => Promise.resolve({ results: [{ hits: [] }] }));
+describe('createInstantSearchManager with multi index', () => {
+  it('updates the store and searches', () => {
+    expect.assertions(5);
```

Comment:
I don't think you need this, since the promise your return resolves after all your expect have been ran.